### PR TITLE
CPLAT-4533: Post-Dart2 cleanup

### PR DESCRIPTION
**Pull request by olesiathoms-wk**
## Motivation
  
We can finally start removing dart1.

DO NOT MERGE waiting for dart_dev 3.0.0 release.

## Changes
  
- updated dependencies
- unnecessary `new` were removed
- "=" now used as the separator before named parameter default values
- `dev.dart` was updated to use `dart_dev v3` and renamed to `config.dart`
- deprecated lint rules were removed
-  `.pub` and `.buildlog` removed from`.gitignore`
- run `.travis.yaml` with dart 2.4.1, dev and stable versions
- add `build_vm_compilers dev_dependency`
- add `build.yaml`

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
        - [ ] CI passes
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/state_machine/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/state_machine/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: build.yaml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+targets:
+  $default:
+    builders:
+      build_vm_compilers|entrypoint:
+        generate_for:
+          include:
+            - "test/**.dart.vm_test.dart"
```
Comment (evanweible-wf) [inline build.yaml]: I know we've been using this in a lot of places, but we could probably shorten these just a bit:

For build_vm_compilers:
```
"test/**.vm_test.dart"
```

For build_web_compilers:
```
"test/**.browser_test.dart"
```

File: dart_test.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+paths:
+  - test/
```
Comment (evanweible-wf) [inline dart_test.yaml]: This is the default paths configuration, so you can omit it altogether.

File: .travis.yml [hunk 3]
```diff
@@ -1,13 +1,24 @@
...
+      name: "SDK: 2.4.1"
+      script:
+        - pub run dart_dev analyze
+        - pub run dart_dev test --release
+    - dart: stable
+      name: "SDK: stable"
+      script:
+        - pub run dart_dev analyze
+        - pub run dart_dev format --check
+        - pub run dart_dev test --release
```
Comment (evanweible-wf) [inline .travis.yml]: In the stable config, let's do one test run in dev mode as well

File: .gitignore [hunk 4]
```diff
@@ -2,8 +2,6 @@
 build/
 packages/
```
Comment (evanweible-wf) [inline .gitignore]: can also remove this

File: build.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    builders:
+      build_vm_compilers|entrypoint:
+        generate_for:
+          include:
+            - "test/**.vm_test.dart"
+      build_web_compilers|entrypoint:
+        generate_for:
+          include:
+            - "test/**.browser_test.dart"
+            - "example/**.dart"
```
Comment (evanweible-wf) [inline build.yaml]: We should be able to get rid of this `generate_for` config altogether. The defaults work fine for this package

**Comment by todbachman-wf** [comment]
@Workiva/release-management-pp

**Pull request merged**